Kowubah/crypto feature

Flyer107 opened the pull request:
A series of changes were made to support the crypt functionality. In terms of design, the once-account for all structures was used.

The view was updated so that users are able to fill out forms.

Following the backend controller logic was implemented to update the database and pass the user data model.

Flyer107 commented:
Integration tests failed. Will need to cover again.

s1ddh4rthc reviewed (changes requested):
Great job with your code - especially the integration tests! You did a great job following the format of the other integration tests and making sure that they weren't excessively wordy. I've left some comments on some things you can fix.

s1ddh4rthc commented:
You mentioned that the integration tests aren't working - the code looks clean, but make sure the tests also are passing.

Flyer107 commented on src/main/java/net/testudobank/MvcController.java:
```diff
@@ -625,7 +649,49 @@ public String submitTransfer(@ModelAttribute("user") User sender) {
+    submitWithdraw(user);
+
+    // Convert to from dollars to ETH
+    double currentEth = user.getEthPrice();
```
Done.

Flyer107 commented on src/main/java/net/testudobank/MvcController.java:
```diff
@@ -625,7 +649,49 @@ public String submitTransfer(@ModelAttribute("user") User sender) {
+
+    // Convert to from dollars to ETH
+    double currentEth = user.getEthPrice();
+    double storeEth = amountToBuyCrypto/currentEth;
```
Thanks, good catch.

Flyer107 commented:
All integration tests pass now. The following changes were made to test the functionality of the feature. 

MvcControllerIntegTest.java:
Fixed the simple crypto deposit test case and added other test cases. The other test cases deal with the simple selling of crypto and how buying and selling is expected to be handled if an overdraft is present. 

MvcControllerIntegTestHelpers.java:
Helper methods were added to get the current price of eth as the integration tests run. In addition, the helper functions to check transactions and insert into the crypto tables were fixed to make sure the correct data type was inputted into the DB

MvcController.java:
The format of eth and how much the user expected to sell and buy were edited, as these parameters are passed along for comparison.

MvcControllerIntegTestHelpers.java:
Further helper methods were added and edited for ease of readability. 

TestudoBankRepository.java:
A try-catch code block was added to view the stack trace. This provides more error messages if values returned from the DB are not accepted and halt the running of tests.
